### DISCUSSION TITLE
(SVG) imported files lose colors

### DISCUSSION THREAD

**Discussion by age3rcm** [opentoonz/opentoonz#6541] [General]
![untitled7__opentoonz_1 2_2018-02-12_09-42-21](https://user-images.githubusercontent.com/36406840/36095548-bcb4730c-0fd9-11e8-96a4-aa097b02faa6.png)

when i import/load a .svg file, the shapes does not come with any color palletes, and lose any contour.
vector file created with adobe illustrator. with svg profile 1.1, also tested profile 1.0

there is another vector file that has been imported, but nothing shows and the strange number of 65534 appears alongside it

**Comment by artisteacher** [comment]
I'm not having this particular issue. What version of OpenToonz are you using and what is your OS?

**Comment by age3rcm** [comment]
i am using opentoonz 1.2
i am using windows 10 64 bits
that particular svg vector file was made with adobe illustrator 2017

**Comment by RodneyBaker** [comment]
Can you share the SVG or another one that behaves the same way?

**Comment by age3rcm** [comment]
[qtmarks.zip](https://github.com/opentoonz/opentoonz/files/1750062/qtmarks.zip)

**Comment by RodneyBaker** [comment]
There's definitely some strangeness going on there.
**You can get rid of the odd appended number via Right Click>AutoRenumber.**
I can confirm an issue with colors/palette coming into OpenToonz. (I need to look further into that)
I don't seem to have any issue with contour.

And of course the file you shared isn't an SVG but an Illustrator (AI) file and as OpenToonz doesn't import those we have to convert that to SVG to test and that could repair something in the process that won't match your setup.

SVGs don't load into github and many other online sites for security reasons.
Kind of defeats the value of using SVG...
Attached please find an SVG created from a PLI file via OpenToonz that should open in OT correctly.
[acorns.zip](https://github.com/opentoonz/opentoonz/files/1750150/acorns.zip)


I would look first to your export settings in Illustrator.
There may be something there that is causing issues.

**Comment by RodneyBaker** [comment]
Okay, another observation.  This time related to Loading vs Importing.
Using the SVG I uploaded here...

If I attempt to Import it into OpenToonz it comes in without fill colors. (Import creates a copy and then reference that copy)

If I load (which references the original image on disk) I get the SVG will correct colors.

That suggests there is something lost in translation between copying the original image and placing it into OpenToonz.

So until such as time as that is fixed.... **in order to retain the colors/palette it may be best to Load an SVG rather than Import it.**  Note that Import is the default we are presented with so we have to change that to Load (See attached image)

![paletteyes_loadsvgdontimport](https://user-images.githubusercontent.com/2576432/36573802-4c3fdc2e-1808-11e8-8be5-cf9c69ab1460.jpg)

**Comment by RodneyBaker** [comment]
As scaling and placement on the canvas can be an issue with SVG import...

For correct scaling it seems to me that 1:100 yields the best results but that may depend on what program you are creating SVG files in.  I recommend creating them in OpenToonz where possible for maximum compatibility with OpenToonz.

I'd be interested to know if anyone else thinks a different scale works better.

**Comment by shun-iwasawa** [comment]
I briefly investigated the issue.
It seems not to be related the load / inport option, as I failed to load color with the load option.

When exporting SVG from Illustrator, please try setting `SVG Options > Advanced Options > CSS Properties` from `Style Elements` to `Style Attributes` .

In my environment it enables OT to load colors properly.

With the default `Style Elements` option, color information is stored in the style sheet which seems not to be available in OT. With the `Style Elements` option, color information is stored in each `<path>` elements and can be loaded in OT.

**Comment by age3rcm** [comment]
[drawings.zip](https://github.com/opentoonz/opentoonz/files/1752858/drawings.zip)

i managed to missread the file's extension because the icons on windows explorer are very look-a-like
here is the .svg from the project folder itself. the problematic ones.

[qtmarks.zip](https://github.com/opentoonz/opentoonz/files/1752889/qtmarks.zip)
this .svg was made via illustrator's "export as" function.
i loaded it on opentoonz, and the result was acceptable, just one missing shape

[qtmarks2.zip](https://github.com/opentoonz/opentoonz/files/1752899/qtmarks2.zip)
then i loaded this .svg file created with "save as" function and following Iwasawa's instructions and...
hey, it worked! it looks correct? what is that curved brown shape? it does not exist if you open the same file in illustrator, nor it was included in the original .ai file. it also appears in the other vector files that does not have colour

but, as another open issue on the list says: imported .svg shapes creates a lot of vector points.
on illustrator it has 3 points, while on opentoonz it shows that it haves 12 points, 4 times more.
this does not happen if the .svg is opened in illustrator

to renember: i used the "load" option instead of "import" option in these 2 cases.

the .svg file contained in the zip folder "qtmarks2.zip" placed on the project through the "import" button brings the the same result as if placed through "load" button

**Comment by sk0tie12** [comment]
I am on a MacOS version 10.13.6 using the most current version of OpenToonz. When I import my SVG (created in Inkscape) to OpenToonz, some of the colors are missing and the SVG does not look the same. I cannot figure out how or why this is happening. Does anyone know?

Here is the SVG in Inkscape:

<img width="1265" alt="Screen Shot 2019-05-12 at 11 27 25 PM" src="https://user-images.githubusercontent.com/50568917/57600285-3de63600-750e-11e9-8b77-fc75e342dc40.png">

Here is the SVG in Safari web browser:

<img width="558" alt="Screen Shot 2019-05-12 at 11 28 00 PM" src="https://user-images.githubusercontent.com/50568917/57600336-64a46c80-750e-11e9-826d-3ab87a98965a.png">

Here is the SVG imported and loaded into OpenToonz:

<img width="885" alt="Screen Shot 2019-05-12 at 11 29 19 PM" src="https://user-images.githubusercontent.com/50568917/57600364-7a199680-750e-11e9-8b8f-f2dfdeed1c08.png">

**Comment by RodneyBaker** [comment]
Can you share the SVG file itself?

**Comment by sk0tie12** [comment]
Thank you. I had to zip the file. Here it is:
[Headb.svg.zip](https://github.com/opentoonz/opentoonz/files/3175395/Headb.svg.zip)

**Comment by artisteacher** [comment]
I wonder if the problem is due to the high number of layers?

**Comment by RodneyBaker** [comment]
//conjecture on//
You might be better off converting the SVG to PNG and then using the PNG.
Opentoonz doesn't directly use SVGs... it converts the SVG image to Opentoonz native PLI (vector) format.  Opentoonz then uses that PLI file.

What you are experiencing appears to be a limitation in how Opentoonz handles SVG images.

Alternately, you might use the original image (presumably a raster/bitmap image) and then convert it to vector in Opentoonz where you can fine tune the conversion to greatest extent.  Then you (theoretically) will have a more optimized vector image/PLI.

**Comment by artisteacher** [comment]
Maybe @Calabaraburus (who is amazing & fixed a couple other svg display issues) could help...

**Comment by sk0tie12** [comment]
Thank you Rodney and artisteacher for the help. I will reach out to @Calabaraburus if I can on this site.

The original image is a PNG that I converted into an SVG in Inkscape. Initially, I was working with the PNG images of the full figure in Opentoonz as raster levels and ran into issues when animating it with the Plastic tool, as the figure’s legs began to warp and distort between frames during a walk cycle. I wanted to try working with the figure in SVG images as vector levels with the hopes of maneuvering it better with the Plastic tool.

I did attempt to covert the raster level to a vector level but lost much of the quality of the image when attempting to convert. I could go back and try that again or attempt to work with animating it again as a raster level.

**Comment by RodneyBaker** [comment]
The Plastic Tool handles vector and raster imagery the same... or for all intents and purposes the same.
So for images with high gradiation of color using the original PNG may be preferred for that reason as well.

The most difficult part of working with raster images for most people when layering (ala rigged characters) is dealing with alpha channel/transparency.

**Comment by AashnaT** [comment]
I am having the same issue as age3rcm. I have just started learning animation on opentoonz and have been trying to load a random file I made in SVG format from Adobe Illustrator. The file doesn't originally have any fill color but it appears all black in OT, whether I load it or import it. I tried changing CSS Properties to both Style Attributes and Style Elements, but the problem remains. 
Please help if you can.

**Comment by RodneyBaker** [comment]
@AashnaT If you can supply an example SVG to test that will help tremendously.

Note that if you SVG relies on CSS or external file in any way this will be a problem in converting as most programs including Opentoonz don't deal with external links in SVG images.  In these cases you'll want to export with options that convert text to vectors, and keep other properties internal.

As has been mentioned before you might have more success exporting to PNG from Illustrator and converting to vectors in Opentoonz via 'Convert to Vector'.

**Comment by AashnaT** [comment]
@RodneyBaker I tried converting the PNG to Vector and it helped. Thank you so much!

**Comment by gzuk78** [comment]
I can confirm that OpenToonz ignores all SVG gradients, i.e. indirect color references. That is true even if the gradient is only a single color ("swatches" in Inkscape).

If you import an SVG image where some shapes have direct fills, and others are references to single-color gradients, then only the shapes with direct fills show up. The others are still there, but totally transparent.

**Comment by jpolonia23** [comment]
This may be of use to someone

If you are trying to import an svg from Illustrator, you should save the file under the following considerations:

![Configuracion_AI_para_SVG_Opentoonz](https://user-images.githubusercontent.com/30080320/110559778-377d7100-811b-11eb-806f-f06311675ba6.PNG)

Work very well...

![12](https://user-images.githubusercontent.com/30080320/110559999-97741780-811b-11eb-88bb-61aba229ba18.PNG)

**Comment by RodneyBaker** [comment]
This is one of many reports that will need to be reviewed after merging #5334.  Some issues relate to SVG but others to PLI.
It is important to understand that at this time the only way Opentoonz supports SVG is through conversion to PLI.

**Comment by RodneyBaker** [comment]
Same issue as #119?

**Comment by age3rcm** [comment]
> Same issue as [#119](https://github.com/opentoonz/opentoonz/issues/119)?

yes

**Comment by RodneyBaker** [comment]
It's something of a toss up whether to keep this report open or the original (#119).
This report contains very useful information and test files.
But then again so does #119.

The original report does contain suggestions for making sure the SVGs are set up optimally for import into Opentoonz.  But so does this report.

For now we'll keep both reports open.
